corehq/apps/app_manager/suite_xml/post_process/resources.py

```diff
@@ -23,18 +23,20 @@ class Meta(object):
 def add_xform_resource_overrides(domain, app_id, pre_to_post_map):
     overrides_by_pre_id = get_xform_resource_overrides(domain, app_id)
     dirty = False
-    for pre, post in pre_to_post_map.items():
-        if pre in overrides_by_pre_id:
-            if post != overrides_by_pre_id[pre].post_id:
-                raise ResourceOverrideError("Cannot change override of {}".format(pre))
+    for pre_id, post_id in pre_to_post_map.items():
+        if pre_id in overrides_by_pre_id:
+            if post_id != overrides_by_pre_id[pre_id].post_id:
+                raise ResourceOverrideError("""
+                    Cannot change override: domain {}, app {}, pre_id {}
+                """.strip().format(domain, app_id, pre_id))
```

Comment:
I think this will only file one error at a time, would it be useful to restructure this so that all errors are returned?

Reply:
Yeah, I'm reworking this now after running into errors locally. Errors in this are pretty bad news - it'd mean a master form id can be mapped to multiple linked form ids, which means the app is subject to the bug that causes crashes on mobile pre-2.48 - but it's possible that there are apps in this state. I'm likely going to take the approach of writing a verification script and running that before deploying this.

Reply:
I read this too quickly and thought it was a comment on the migration. Anyway, 8fb2505d205 pretty much deals with this.